ddtrace/contrib/starlette/patch.py

```diff
@@ -10,16 +12,58 @@
 log = get_logger(__name__)
 
 config._add(
-    "starlette", dict(_default_service="starlette", request_span_name="starlette.request", distributed_tracing=True)
+    "starlette",
+    dict(
+        _default_service="starlette",
+        request_span_name="starlette.request",
+        distributed_tracing=True,
+        aggregate_resources=True,
+    ),
 )
 
 
+def traced_init(wrapped, instance, args, kwargs):
+    global routes
+    routes = kwargs["routes"]
+    mw = kwargs.pop("middleware", [])
+    mw.insert(0, Middleware(TraceMiddleware, integration_config=config.starlette))
+    kwargs.update({"middleware": mw})
+
+    wrapped(*args, **kwargs)
+
+
+def get_resource(scope):
+    path = None
+    global routes
+    for route in routes:
+        match, _ = route.matches(scope)
+        if match == Match.FULL:
+            path = route.path
+            break
+        elif match == Match.PARTIAL and path is None:
+            path = route.path
+    return path
+
+
+def call_decorator(func):
+    def call_wrapper(*args, **kwargs):
+        resource = get_resource(args[1])
+        if config.starlette["aggregate_resources"]:
+            args[1]["resource"] = resource
```

Comment:
I am wary of adding our own attributes to the ASGI `scope` object. There has been some discussion of adding a `scope["route"]` for the same purpose we have in https://github.com/encode/starlette/pull/804. Being that we can do this do this by passing a function to the `TraceMiddleware`, I think we should avoid changing the `scope` object.

Reply:
The approach could be similar to https://github.com/open-telemetry/opentelemetry-python/blob/master/instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py#L59